tada_a_bigoh.py

```diff
@@ -120,7 +130,62 @@
+        print(analysis.analyze_big_oh(ratio))
+        if indicator < 0.1:
+            constants.QUIT_BY_INDICATOR = 1
+        # store indicator
+        constants.INDICATOR_VALUE = constants.INDICATOR
+        # store runningtime
+        constants.TOTAL_RUNNING_TIME = time.time() - start_time
+        last_bench_meta = current_benchmark.get_metadata()
+        name = last_bench_meta["name"]
+        # store benchmark metadata
+        constants.NAME_OF_EXPERIMENT = configuration.get_experiment_info(vars(tada_arguments))
+        constants.CPU_TYPE = last_bench_meta["cpu_model_name"]
+        constants.OS = last_bench_meta["platform"]
+        constants.PYTHON_VERSION = last_bench_meta["python_version"]
+        # store run metadata
+        with open("_results" + constants.SEPARATOR + name + constants.JSON_EXT, 'r') as f:
+            readlastjson = json.load(f)
+        last_exp_run_metadata = readlastjson["benchmarks"][0]["runs"][0]["metadata"]
+        constants.CPU_TEMP = last_exp_run_metadata["cpu_temp"]
+        constants.MEM_MAX_RSS = last_exp_run_metadata["mem_max_rss"]
+        for item in total_loop_list:
+            sum_of_loops += item
+        # calculate avg total loops
+        constants.PYPERF_AVG_EXPERIMENT_ROUNDS = sum_of_loops / len(total_loop_list)
+        # calculate last two loop growth ratio
+        if len(total_loop_list) >= 2:
+            constants.PYPERF_LAST_TWO_EXPERIMENT_ROUNDS = total_loop_list[-1] / total_loop_list[-2]
+        # check if result is expected
+        if tada_arguments.expect is not None and tada_arguments.expect in analysis.analyze_big_oh(ratio):
+            constants.RESULT = 1
+        df = pd.read_csv("experiment_data.csv")
+        # EXPERIMENT_RELIABILITY, CPU_TYPE, CPU_TEMP, TOTAL_RUNNING_TIME, QUIT_BY_MAX_RUNTIME, QUIT_BY_INDICATOR, QUIT_BY_BACKFILL, MEM_MAX_RSS, OS, INDICATOR_VALUE, BACKFILL_TIMES, PYPERF_AVG_EXPERIMENT_ROUNDS, NAME_OF_EXPERIMENT
+        df_new = pd.DataFrame(
+            {
+                "EXPERIMENT_RELIABILITY": constants.RESULT,
+                "CPU_TYPE": constants.CPU_TYPE,
+                "CPU_TEMP": constants.CPU_TEMP,
+                "TOTAL_RUNNING_TIME": constants.TOTAL_RUNNING_TIME,
+                "QUIT_BY_MAX_RUNTIME": constants.QUIT_BY_MAX_RUNTIME,
+                "QUIT_BY_INDICATOR": constants.QUIT_BY_INDICATOR,
+                "QUIT_BY_BACKFILL": constants.QUIT_BY_BACKFILL,
+                "MEM_MAX_RSS": constants.MEM_MAX_RSS,
+                "OS": constants.OS,
+                "INDICATOR_VALUE": constants.INDICATOR_VALUE,
+                "BACKFILL_TIMES": constants.BACKFILL_TIMES,
+                "PYPERF_AVG_EXPERIMENT_ROUNDS": constants.PYPERF_AVG_EXPERIMENT_ROUNDS,
+                "PYPERF_LAST_TWO_EXPERIMENT_ROUNDS_RATIO": constants.PYPERF_LAST_TWO_EXPERIMENT_ROUNDS_RATIO,
+                "NAME_OF_EXPERIMENT": constants.NAME_OF_EXPERIMENT,
+                "PYTHON_VERSION": constants.PYTHON_VERSION,
```

Comment:
Could also add an attribute for the data generation strategy since we have three different approaches now. It could be read in from the argument `--types`

Reply:
Good suggestion, I'll add this into the variable storage feature